src/app/features/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { Action } from '@ngrx/store';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { initialAppState } from '@store/index';
+import { of, ReplaySubject } from 'rxjs';
+import { HydrateNewVehicleRecordComponent } from './hydrate-new-vehicle-record.component';
+import { createVehicleRecord, createVehicleRecordSuccess } from '@store/technical-records';
+
+describe('HydrateNewVehicleRecordComponent', () => {
+  let component: HydrateNewVehicleRecordComponent;
+  let fixture: ComponentFixture<HydrateNewVehicleRecordComponent>;
+  let techRecordService: TechnicalRecordService;
+  let actions$ = new ReplaySubject<Action>();
+  let errorService: GlobalErrorService;
+  let route: ActivatedRoute;
+  let router: Router;
+  let store: MockStore;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [HydrateNewVehicleRecordComponent],
+      providers: [provideMockActions(() => actions$), provideMockStore({ initialState: initialAppState })],
+      imports: [HttpClientTestingModule, RouterTestingModule]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(HydrateNewVehicleRecordComponent);
+    route = TestBed.inject(ActivatedRoute);
+    errorService = TestBed.inject(GlobalErrorService);
+    router = TestBed.inject(Router);
+    store = TestBed.inject(MockStore);
+    techRecordService = TestBed.inject(TechnicalRecordService);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('get Vehicle', () => {
+    expect(component.vehicle$).toBeTruthy();
+  });
+
+  it('should navigate away on no data', () => {
+    const navigateSpy = jest.spyOn(router, 'navigate').mockImplementation(() => Promise.resolve(true));
+    jest.spyOn(component, 'vehicle$', 'get').mockImplementation(() => of(undefined));
+    component.ngOnInit();
+    expect(navigateSpy).toBeCalledWith(['..'], { relativeTo: route });
+  });
```

Comment:
```suggestion
  it('should navigate away on no data', () => {
    jest.spyOn(techRecordService, 'editableVehicleTechRecord$').mockReturnValue(of(undefined));
    const navigateSpy = jest.spyOn(component, "navigateBack")
    component.vehicle$
    expect(navigateSpy).toBeCalledTimes(1);
  });
```
I don't actually know if the suggested test passes. But this test in confusing me:
- The component does not have an onInit lifecycle hook, so I don't know why it's being called
- If you're testing behaviour from the getter, mocking said getter in the test doesn't really make sense.

Reply:
Done, good spot.